### PR TITLE
Improve error handling for TableWorkspace

### LINKED ISSUE #22918: Mixing up the TableWorkspace column type and value should throw error.
### Expected behavior
Mixing up the column type and column name in the table workspace python api should show an error in the script window output

### Actual behavior
An error is shown in the sample logs, but not in the script window. Depending on your window configuration it appears to silently fail.

### Steps to reproduce the behavior
```
from mantid.api import WorkspaceFactory
tbl = CreateEmptyTableWorkspace()

tbl.addColumn("double", "k")
tbl.addColumn("double", "h")

# This creates an error in the log, but not the script window
tbl.addColumn("l", "double")
```

### Platforms affected
All

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of work.**
Primarily throwing exceptions rather than logging when there is an error.
This reports better to python, and prevents silent problems.

**Report to:** Sam Jackson

**To test:**

1.  All unit and sytem tests pass
2. Code review.
3. The last line of this shold return a useful error.
```python

from mantid.api import WorkspaceFactory
tbl = CreateEmptyTableWorkspace()

tbl.addColumn("double", "k")
tbl.addColumn("double", "h")

tbl.addColumn("l", "double")
```

Fixes #22918. 

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by KarlPalmen — APPROVED** [review]
Works as described. Errors transferred to python page.

File: Framework/WorkflowAlgorithms/src/MuonGroupAsymmetryCalculator.cpp [hunk 1]
```diff
@@ -130,7 +130,6 @@ MuonGroupAsymmetryCalculator::estimateAsymmetry(const Workspace_sptr &inputWS,
     table->addColumn("double", "norm");
     table->addColumn("str", "name");
     table->addColumn("str", "method");
-    table->removeRow(0);
```
Comment (SimonHeybrock) [inline Framework/WorkflowAlgorithms/src/MuonGroupAsymmetryCalculator.cpp]: Why is this line removed?
